### PR TITLE
Hyphy version 2.3.14 and mac

### PR DESCRIPTION
**Pull request by rdvelazquez**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

2.3.14 doesn't use HYPHYMPI (HYPHYMPI is really only used for GARD and 2.3.14 doesn't have a stable version of GARD); because no HYPHYMPI, 2.3.14 doesn't need openmpi

### CODE REVIEW

File: recipes/hyphy/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+c_compiler:
```
Comment (bgruening) [inline recipes/hyphy/conda_build_config.yaml]: Any reason this file and the old compilers are needed?

Comment (rdvelazquez) [reply recipes/hyphy/conda_build_config.yaml]: Thanks for looking at this @bgruening

Not sure if this file is still needed, just following what was done in the past. 

The old compilers shouldn't be needed.

@davebx any thoughts?

Comment (davebx) [reply recipes/hyphy/conda_build_config.yaml]: I honestly don't remember adding that. I say remove it and see what happens.

**Comment by rdvelazquez** [comment]
Do I need to do anything else to get `bioconda-test` to run?

**Comment by rdvelazquez** [comment]
Having Circle-CI enabled on my fork may have been causing the CI to hang. I've deactivated CI on my fork and pushed https://github.com/bioconda/bioconda-recipes/pull/14382/commits/2230773237bfb96cd7b67ec03dbe070b2a619a0b to re-trigger a re-build.

**Comment by bgruening** [comment]
Thanks @rdvelazquez

**Pull request merged**